modules/openapi-generator/src/main/resources/typescript-nestjs/api.module.mustache

```diff
@@ -0,0 +1,40 @@
+import { DynamicModule, HttpService, HttpModule, Module, Global } from '@nestjs/common';
+import { Configuration } from './configuration';
+import { BASE_PATH } from './variables';
+
+{{#apiInfo}}
+{{#apis}}
+import { {{classname}} } from './{{importPath}}';
+{{/apis}}
+{{/apiInfo}}
+
+@Global
+@Module({
+  imports:      [ HttpModule ],
+  exports:      [   
+    {{#apiInfo}}{{#apis}}{{classname}}{{#hasMore}},
+    {{/hasMore}}{{/apis}}{{/apiInfo}} 
+  ],
+  providers: [
+    {
+        provide: 'BASE_PATH',
+        useValue: BASE_PATH
+    },
+    {{#apiInfo}}{{#apis}}{{classname}}{{#hasMore}},
+    {{/hasMore}}{{/apis}}{{/apiInfo}} 
+  ]
+})
+export class ApiModule {
+    public static forRoot(configurationFactory: () => Configuration): DynamicModule {
+        return {
+            module: ApiModule,
+            providers: [ { provide: Configuration, useFactory: configurationFactory } ]
```

Comment:
Same as (1.) here: if this is just named `Configuration`, won't the injection break if you import multiple generated modules for different APIs? Also, what if you already have a service named `Configuration` in your app (which seems likely), won't this break the DI?

Reply:
If classes are defined in different files (even with the same class name) - these are distinct class references from the JS perspective.
So if Nestjs uses this reference as an injection token (rather than, say, the string name of the class, then there should be no conflicts for DI.
E.g. Angular uses actual class reference for DI, not a name - so it's not an issue there.

Would you be willing to investigate how Nestjs DI works and post a comment in this PR? It can help with defining the direction for further changes

Reply:
@quezak No it will not break injection